kafka/adminapi.go

```diff
@@ -103,7 +104,23 @@ import "C"
 // AdminClient is derived from an existing Producer or Consumer
 type AdminClient struct {
 	handle    *handle
-	isDerived bool // Derived from existing client handle
+	isDerived bool   // Derived from existing client handle
+	isClosed  uint32 // to check if Admin Client is closed or not.
+}
+
+// IsClosed returns boolean representing if client is closed or not
+func (a *AdminClient) IsClosed() bool {
+	if a.isClosed == 0 {
```

Comment:
```suggestion
	return atomic.LoadUint32(&a.isClosed) == 1
```

also for other `IsClosed()`